src/libraries/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs

```diff
@@ -107,6 +107,98 @@ public void CheckEndReachableDoesNotCrossPastEnd()
 
         #endregion
 
+        #region Offset
+
+        [Fact]
+        public void GetOffset_SingleSegment()
+        {
+            var buffer = new ReadOnlySequence<T>(new T[50]);
+            Assert.Equal(25, buffer.GetOffset(buffer.GetPosition(25)));
+        }
+
+        private (BufferSegment<T> bufferSegment1, BufferSegment<T> bufferSegment4) GetBufferSegment()
+        {
+            var bufferSegment1 = new BufferSegment<T>(new T[50]);
+            BufferSegment<T> bufferSegment2 = bufferSegment1.Append(new T[50]);
+            BufferSegment<T> bufferSegment3 = bufferSegment2.Append(new T[0]);
+            return (bufferSegment1, bufferSegment3.Append(new T[50]));
+        }
+
+        private ReadOnlySequence<T> GetFourSegmentsReadOnlySequence()
+        {
+            (BufferSegment<T> bufferSegment1, BufferSegment<T> bufferSegment4) = GetBufferSegment();
+            return new ReadOnlySequence<T>(bufferSegment1, 0, bufferSegment4, 50);
+        }
+
+        [Fact]
+        public void GetOffset_MultiSegment_FirstSegment()
+        {
+            ReadOnlySequence<T> buffer = GetFourSegmentsReadOnlySequence();
+            Assert.Equal(25, buffer.GetOffset(buffer.GetPosition(25)));
```

Comment:
Can we add a test that enumerates through all offsets from 0 to `buffer.Length` and makes sure that `buffer.GetOffset(buffer.GetPosition(i))` always returns `i`.

Reply:
Another test to validate behavior is to make sure this always holds true: `sequence.Slice(0, pos).Length == sequence.GetOffset(pos)`

Reply:
done